File: core/scripts/Java-Library/src/chh/math/geo/Volume.java
package chh.math.geo;

/**
 * A wrapper class representing a three-dimensional measurement. Values may be {@code double}s.
 * @author  Christopher H. Harvey <chrisharvey2pi@gmail.com>
 * @version 2014.03.29
 */
public class Volume {
	/** The value of this volume. */
	private final double value;
	/**
	 * Constructs a new Volume object.
	 * @param x the volume
	 */
	public Volume(double x) {
		this.value = x;
	}

	/**
	 * Returns the value of this volume.
	 * @return the value of this volume.
	 */
//	public final double getValue() {
//		return this.value;
//	}
	
	/**
	 * Multiplies a length by an area and returns a volume.
	 * @param a the length
	 * @param b the area
	 * @return a volume equal to the product of both lengths
	 */	
	public static final Volume multiply(Length a, Area b) {
		return new Volume(a.value * b.value);
	}
	/**
	 * Multiplies an area by a length and returns a volume.
	 * Equivalent to {@see multiply(Length, Area)}.
	 * @param a the area
	 * @param b the length
	 * @return a volume equal to the product of a and b
	 */	
	public static final Volume multiply(Area a, Length b) {
		return Volume.multiply(b, a);
	}
}

File: core/scripts/Volume.js
/**
  * A wrapper class representing a three-dimensional measurement.
  * @param x this Length object represented as a non-negative number
  * @param u optional: a string representing the units of measurement
  */
function Volume(x, u) {
  this.value = +x || 0;
  this.units = '' + (u || '');
}

/**
  * Returns a string representation of this object.
  */
Volume.prototype.toString = function () {
  return '' + this.value + this.units + '³'; // U+00B3
}

/**
  * Adds another Volume object to this Volume object.
  * @param volume the Volume object to add to this object
  * @return       a Volume equal to the sum of both Volumes
  */
Volume.prototype.add = function (volume) {
  if (this.units === volume.units) return new Volume(this.value + volume.value, this.units);
  else                             return '' + this + ' + ' + volume;
}

/**
  * Returns a scale factor of this Volume object.
  * If the scale factor is <1, returns a new Area "smaller" than this Volume.
  * If the scale factor is >1, returns a new Area "bigger"  than this Volume.
  * If the scale factor is =1, returns a new Area equivalent to  this Volume.
  * @param k a non-negative number: the scale factor
  * @return  the product of this Volume and the scale factor
  */
Volume.prototype.scale = function (k) {
  return new Volume(this.value * k, this.units);
}

Volume.prototype.multiply = function (length) { return null; }
